### PR TITLE
Move Email Management to `/email` route

### PR DESCRIPTION
**Pull request by cbauerman**
Some users wish to buy or manage G Suite without necessarily buying or managing their domain on WordPress.com. Currently G Suite is only accessible via (and nested under) the Domains management screens and paths, i.e. `/domains/manage/email/foo.com`. 

We also check that you have a domain when a user tries to visit any of the email management screens directly.

This patch moves email management to its own top-level `/email/foo.com` path and allows it to work without a domain. No other functionality should be affected.

#### Changes

* Email is now managed from `/email` routes. A mostly complete list of these changes are below:
  * `domains/manage/email` *->* `/email`
  * `domains/manage/email/:site` *->* `/email/:site`
  * `domains/manage/add-gsuite-users/:site` ->`email/add-gsuite-users/:site`
  * `domains/manage/:domainName/add-gsuite-users/:site` ->`email/:domainName/add-gsuite-users/:site`
  * `domains/manage/email/:domainName/edit/:site` *->* `/email/:domainName/manage/:site`

* Email has been separated from the Domains Section and given it's own Email Section
* Miscellaneous fixes to CSS names and opportunistic updates from `google-apps` -> `gsuite`

#### Testing Instructions

*Prerequisites*
* A site with a domain and GSuite
* A site with a domain and without GSuite ( can be the same as the above site )  
* A site without a domain

*Testing New Routes*
1. Navigate to `/email/:site` for your site with GSuite
1. Confirm that the page is unchanged ( see screenshot below for reference, though note you may or may not have the same notices ) screen shot 2019-02-19 at 2 36 00 pm
1. Click one of the "Add GSuite Users" Button
1. Observe the new URL. Confirm it matches `/email/:domain/add-gsuite-users/:site`
1. Observe the page and confirm it is unchanged ( screenshot below ) screen shot 2019-02-19 at 3 37 09 pm
1. Click the "Back" Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Navigate to `/email/:domain/forwarding/:site` for the site with a domain and without g Suite
1. Confirm the Email Forwarding page is uncharged ( screenshot below ) screen shot 2019-02-19 at 3 39 55 pm
1. Test Email Forwarding with the following Instructions
   1. Add email forward
   1. Observe that email forward appears at top
   1. Click mail icon to resend verification email
   1. Verify forward in your email
   1. Refresh, observe that email icon is gone
   1. Click delete icon
   1. Observe that email forward disappear
1. Click the Back Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Confirm that you are shown the G Suite CTA and that it is unchanged  screen shot 2019-02-19 at 3 41 31 pm
1. Click "Add G Suite"
1. Confirm you are taken to the add G Suite users page again
1. Fill in the info and click continue
1. Confirm G Suite was correctly added to your cart

*Testing Legacy Redirects*
1. Navigate to `/domains/manage/email`
1. Confirm you are taken to `/email` and given a list of sites ( if you have more than one)
1. Navigate to `/domains/manage/email/:site` for your site with G Suite
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/email/:domain/edit/:site`
1. Confirm you are taken to `/email/:domain/manage/:site`
1. Navigate to `/domains/manage/:domain/email-forwarding/:site
1. Confirm you are taken to `/email/:domain/forwarding/:site`
1. Navigate to `/domains/manage/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/:domain/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:domain/add-gsuite-users/:site`

### CODE REVIEW

File: client/my-sites/controller.js [hunk 1]
```diff
@@ -156,16 +158,16 @@ function renderSelectedSiteIsDomainOnly( reactContext, selectedSite ) {
 	makeLayout( reactContext, noop );
 	clientRender( reactContext );
 }
-
+// TODO
```
Comment (gravityrail) [inline client/my-sites/controller.js]: TODO: Please remove this TODO :)

Comment (cbauerman) [reply client/my-sites/controller.js]: TODO: ✅

File: client/my-sites/email/email-forwarding/email-forwarding-details.jsx [hunk 2]
```diff
@@ -16,13 +16,16 @@ import { localize } from 'i18n-calypso';
...
 
+// eslint-disable-next-line react/prefer-es6-class
 const EmailForwardingDetails = createReactClass( {
 	displayName: 'EmailForwardingDetails',
 	mixins: [ analyticsMixin( 'domainManagement', 'emailForwarding' ) ],
 
 	render: function() {
 		return (
 			<p className="email-forwarding__explanation">
+				{' '}
```
Comment (gravityrail) [inline client/my-sites/email/email-forwarding/email-forwarding-details.jsx]: Leftover prettify artifact

File: client/my-sites/email/email-forwarding/email-forwarding-limit.jsx [hunk 3]
```diff
@@ -23,6 +23,8 @@ class EmailForwardingLimit extends React.Component {
 
 		return (
 			<div className="email-forwarding__limit">
+				{' '}
```
Comment (gravityrail) [inline client/my-sites/email/email-forwarding/email-forwarding-limit.jsx]: Leftover prettify artifact

File: client/my-sites/email/email-forwarding/index.jsx [hunk 4]
```diff
@@ -33,19 +33,20 @@ class EmailForwarding extends React.Component {
 		}
 		return (
 			<Main className="email-forwarding">
+				{' '}
```
Comment (gravityrail) [inline client/my-sites/email/email-forwarding/index.jsx]: Leftover prettify artifact

File: client/my-sites/email/email-forwarding/index.jsx [hunk 5]
```diff
@@ -33,19 +33,20 @@ class EmailForwarding extends React.Component {
...
+				{ /* eslint-disable-line wpcalypso/jsx-classname-namespace */ }
 				<Header onClick={ this.goToEditEmail } selectedDomainName={ this.props.selectedDomainName }>
 					{ this.props.translate( 'Email Forwarding' ) }
 				</Header>
-
 				<SectionHeader label={ this.props.translate( 'Email Forwarding' ) } />
-				<Card className="email-forwarding-card">
+				<Card className="email-forwarding__card">
+					{' '}
```
Comment (gravityrail) [inline client/my-sites/email/email-forwarding/index.jsx]: Leftover prettify artifact

**Comment by cbauerman** [comment]
Closing in favor of smaller PRs

**Pull request closed without merging**